src/cache/encryption.rs

```diff
@@ -1,4 +1,6 @@
-#![allow(clippy::type_complexity)] // the static types are too long, this cases no compilation problems though.
+/// --- Add appropriate documentation for this type ---
```

Comment:
One final minor suggestion I would like to make, I would suggest updating this with an appropriate documentation.

Reply:
All of that is getting resolved by the next commit